### DISCUSSION TITLE
External Dependency Dashboard

### DISCUSSION THREAD

**Discussion by amezin** [renovatebot/renovate#28900] [Suggest an Idea]
### Tell us more.

Renovate does not support dependency dashboard on multiple platforms. In some cases, it's just not implemented (yet?), in other cases there are platform limitations. Also, maybe you just don't want a special issue in your repository and would prefer a different UI.

I'm using Renovate with Bitbucket Server - there seems to be no built-in "issues" (only Jira), also no markdown checkboxes. And I'd definitely not want the dashboard in Jira.

Instead, I can publish arbitrary HTML on Jenkins (and, for example, add links that trigger new jobs into it). I would be able to implement my own dashboard, if Renovate wrote all necessary information (detected dependencies, pending PRs, warnings, etc.) for all processed repos to a `.json` file (could it be called a "report"?).

Ideally, Renovate should also have a flag to avoid its dependency dashboard creation (to get rid of warnings in the log on unsupported platforms, and, maybe, get rid of the unwanted issue on supported ones), but still keep things like `dependencyDashboardApproval` working.

I'm willing to work on this feature myself - I only need an approval and maybe a suggestion on the format.

**Comment by secustor** [comment]
;) It is even called report: https://docs.renovatebot.com/self-hosted-configuration/#reporttype

> Ideally, Renovate should also have a flag to avoid its dependency dashboard creation (to get rid of warnings in the log on unsupported platforms, and, maybe, get rid of the unwanted issue on supported ones), but still keep things like dependencyDashboardApproval working.

This will take some discussion and bigger refactoring as the Dashboard are at the moment tight to the platform. 
TBH not sure if we want to go there as it would mean a significant additional maintenance effort.

**Reply by amezin** [reply]
Hm, can't `ensureIssue`/`ensureIssueClosing` in [`dependency-dashboard.ts`](https://github.com/renovatebot/renovate/blob/main/lib/workers/repository/dependency-dashboard.ts) simply be skipped (just like when `dryRun` is set)?

**Reply by secustor** [reply]
You can simply turn off the dependencyDashboard, tough this will not give the desired approval workflow. https://docs.renovatebot.com/configuration-options/#dependencydashboard

**Reply by amezin** [reply]
If I'm running Renovate with a platform that doesn't support issues (Bitbucket Server), are approvals expected to work through `checkedBranches`?

With `dependencyDashboard` enabled but obviously not working.

**Reply by secustor** [reply]
Not sure TBH, I never had this scenario, but you could try to run with `checkedBranches` targeting the approval branches and overwriting `dependencyDashboardApproval: null` to remove the setting.

**Comment by jamietanna** [comment]
I've written up how to do this [here](https://www.jvt.me/posts/2025/05/30/renovate-dependency-dashboardless/)

Mend Renovate Cloud, and Mend Renovate Self-Hosted follow a similar (but not exact) approach to do similar functionality.

The `checkedBranches` is what you'd want to use here, yes
